### DISCUSSION TITLE
A Very Starange Behavior from CloudFlare IPs

### DISCUSSION THREAD

**Discussion by MEM-917** [XTLS/Xray-core#3657] [General]
Hi

I Have 2 different personal VPSes in the US and Austria with all possible different protocols.(Reality, WebSocket, gRPC, OpenVPN, Hysteria, and ....). Everything was all good until I checked the logs and saw that some WS connections had IPs that didn't originate from Iran(where all the users live) and belonged to NETCUP ISP.
After some digging, I noticed the problem is the source address. E.g. connections with the CloudFlare proxied domains and **www[.]speedtest[.]net** all the logs are as they should be with true source IP(client real IP) but when the address is **sourceforge[.]net** the registered IPs in the logs are all over the place(sometimes more than 20 IPs all from netcup).

Has anyone encountered this problem?(especially from other countries)

p.s. The problem is not WebSocket specific and gRPC has the same issue.

**Comment by underdog-03** [comment]
What is the configuration for IP address spoofing and server routing?

Here are a few possibilities worth investigating.

Sourceforge.net's CDN or Proxy Setup: Sourceforge.net might be using a CDN or proxy setup that's causing WebSocket connections to be routed through different IPs, leading to the NETCUP ISP IPs appearing in your logs.

IP Spoofing or Misrouting: It's possible that someone is spoofing or misrouting WebSocket connections to sourceforge.net, causing incorrect IPs to show up in your logs. (Unfortunately, this is a possibility given the current state of the Iranian network these days)

Network Routing Issues: There might be issues with the network routing between your VPSes and sourceforge.net, which could cause WebSocket connections to take an unexpected path, resulting in the incorrect IPs. While this is unlikely, it's still a potential cause worth checking
